modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/FullContentNodeTranslator.java

```diff
@@ -0,0 +1,127 @@
+package com.enonic.xp.core.impl.content;
+
+import java.util.function.Function;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.enonic.xp.content.Content;
+import com.enonic.xp.content.ContentId;
+import com.enonic.xp.content.ContentNotFoundException;
+import com.enonic.xp.content.ContentPath;
+import com.enonic.xp.content.Contents;
+import com.enonic.xp.context.ContextAccessor;
+import com.enonic.xp.core.impl.content.serializer.ContentDataSerializer;
+import com.enonic.xp.node.Node;
+import com.enonic.xp.node.NodePath;
+import com.enonic.xp.node.NodeService;
+import com.enonic.xp.node.Nodes;
+
+public class FullContentNodeTranslator
```

Comment:
this class looks like ContentNodeTranslator duplicate with minor changes. Can it be one class with `if` instead?